src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/KestrelHttpParser.cs

```
@@ -109,6 +109,11 @@ public KestrelHttpParser(IKestrelTrace log)
 
                     pathEncoded = true;
                 }
+                else if (ch == 0)
```

Fail fast. The null character is never valid in the start line.